tardis/io/tests/test_config_reader.py

```diff
@@ -68,3 +71,78 @@ def test_config_hdf(hdf_file_path, tardis_config_verysimple):
     actual = pd.read_hdf(hdf_file_path, key="/simulation/config")
     expected = expected.get_properties()["config"]
     assert actual[0] == expected[0]
+
+
+def test_model_section_config(tardis_config_verysimple):
+    conf = Configuration.from_config_dict(
+        tardis_config_verysimple, validate=True, config_dirname="test"
+    )
+
+    assert conf.model.structure.density.type == "branch85_w7"
+
+    tardis_config_verysimple["model"]["structure"]["velocity"][
+        "start"
+    ] = "2.0e4 km/s"
+    tardis_config_verysimple["model"]["structure"]["velocity"][
+        "stop"
+    ] = "1.1e4 km/s"
+    with pytest.raises(ValueError) as ve:
+        if (
+            conf.model.structure.velocity.start
+            < conf.model.structure.velocity.stop
+        ):
+            raise ValueError("Stop Value must be greater than Start Value")
+    assert ve.type is ValueError
+
+
+def test_supernova_section_config(tardis_config_verysimple):
+    conf = Configuration.from_config_dict(
+        tardis_config_verysimple, validate=True, config_dirname="test"
+    )
+    tardis_config_verysimple["supernova"]["time_explosion"] = "-10 day"
+    tardis_config_verysimple["supernova"][
+        "luminosity_wavelength_start"
+    ] = "15 angstrom"
+    tardis_config_verysimple["supernova"][
+        "luminosity_wavelength_end"
+    ] = "0 angstrom"
+    with pytest.raises(ValueError) as ve:
+        if conf.supernova.time_explosion.value > 0:
+            raise ValueError("Time of Explosion cannot be negative")
+    assert ve.type is ValueError
+
+    with pytest.raises(ValueError) as ve:
+        if (
+            conf.supernova.luminosity_wavelength_start.value
+            < conf.supernova.luminosity_wavelength_end.value
+        ):
+            raise ValueError(
+                "End Limit must be greater than Start Limit for Luminosity"
+            )
+    assert ve.type is ValueError
+
+
+def test_plasma_section_config(tardis_config_verysimple):
+    conf = Configuration.from_config_dict(
```

Comment:
Please add a docstring describing this test

Reply:
Sure :)